Driver generate corrupted beacon (invalid TLV elements) during CSA
on 2.4 GHz 40 MHZ-wide channel. This patch fixes that corruption.

--- a/bcmdrivers/broadcom/net/wl/impl69/main/src/wl/sys/wlc_vht.c
+++ b/bcmdrivers/broadcom/net/wl/impl69/main/src/wl/sys/wlc_vht.c
@@ -1857,23 +1857,13 @@ wlc_vht_ie_len_csa_wrapper(wlc_bsscfg_t
 
 	chspec = wlc_csa_get_csa_chspec(cfg->wlc->csa, cfg);
 
-	ielen = TLV_HDR_LEN + sizeof(dot11_vht_transmit_power_envelope_ie_t);
+	ielen = TLV_HDR_LEN;
 
 	/* wb_csa_ie not present in 2.5/5/10/20MHz channels */
 	if (!CHSPEC_IS20(chspec)) {
 		ielen += sizeof(dot11_wide_bw_chan_switch_ie_t);
 	}
 
-	/* vht transmit power envelope IE length depends on channel width,
-	 * update channel wrapper IE length
-	 */
-	if (CHSPEC_IS40(chspec)) {
-		ielen += 1;
-	} else if (CHSPEC_IS80(chspec)) {
-		ielen += 2;
-	} else if (CHSPEC_IS8080(chspec) || CHSPEC_IS160(chspec)) {
-		ielen += 3;
-	}
 	return ielen;
 }
 
